### ISSUE TITLE
Removing view does not remove ExoPlayer from layout

### ISSUE THREAD

**Issue by zishanj** [google/ExoPlayer#2647]
I am adding `SimpleExoPlayerView` programmatically during runtime and removing it later using `removeViewAt`. The problem is, it does not get completely removed from the layout. I have checked the debug and after `removeViewAt` it completely removes it from the memory, but it does not seems to refresh the layout automatically. If I use `removeAllViews`, then it removes all other views and refreshes the layout automatically, but the `SimpleExoPlayerView` remains there.

How to resolve it?

**Comment by ojw28** [comment]
Does the same issue occur if you add and remove a regular `Surface` instead of `SimpleExoPlayerView`? If so this is not an ExoPlayer issue, and so I'd recommend filing a bug on the Android issue tracker and/or asking on StackOverflow.

Aside: Please fill in the issue template properly. It's unclear whether this is device specific, for example. Thanks!

**Comment by zishanj** [comment]
The only temporary solution that exists is to hide it.

P.S This issue is not device specific.

**Comment by ojw28** [comment]
I made a minimal change to the demo app to test this. The change I made was to `PlayerActivity`. I stashed `rootView` as a member variable and modified `onClick` to remove the view using `rootView.removeViewAt(rootView.indexOfChild(simpleExoPlayerView));` if one of the buttons is pressed. I also gave `rootView` a bright background color so it's obvious when views above it have been removed properly.

Everything works as expected when I do this, so closing the issue as not reproducible. It doesn't really sound like an ExoPlayer issue anyway. If you want us to investigate further you'll need to file a new issue that properly fills out the issue template, including clear reproduction steps as to how this can be reproduced in the demo app. If you cannot reproduce the issue by making minimal modifications to the demo app, it's likely a bug in your application code.
